xtools build hard-coded to use IAMCU compiler for all x86 targets

I have brew binutils 2.30 installed, but it installs it as "greadelf" and not "readelf" and CMake can't find it. Creating a symlink did not help.

`
CMake Error at /Users/andrew/zephyr/cmake/extensions.cmake:984 (message):
  No such file or directory: CMAKE_READELF: 'CMAKE_READELF-NOTFOUND'
Call Stack (most recent call first):
  /Users/andrew/zephyr/cmake/compiler/gcc.cmake:14 (assert_exists)
  /Users/andrew/zephyr/cmake/toolchain.cmake:38 (include)
  /Users/andrew/zephyr/cmake/app/boilerplate.cmake:242 (include)
  CMakeLists.txt:1 (include)`

We could improve the error message too.

"CMake Error at /Users/andrew/zephyr/cmake/extensions.cmake:984 (message): No such file or directory: CMAKE_READELF: 'CMAKE_READELF-NOTFOUND'"

Actually means that the build can't find the toolchain.

--- a/cmake/toolchain/xtools.cmake
+++ b/cmake/toolchain/xtools.cmake
@@ -6,8 +6,13 @@ set(TOOLCHAIN_HOME ${XTOOLS_TOOLCHAIN_PATH})
 
 set(COMPILER gcc)
 
-set(CROSS_COMPILE_TARGET_arm arm-none-eabi)
-set(CROSS_COMPILE_TARGET_x86 i586-zephyr-elfiamcu)
+set(CROSS_COMPILE_TARGET_arm arm-zephyr-eabi)
+set(CROSS_COMPILE_TARGET_x86 i586-zephyr-elf)
+set(CROSS_COMPILE_TARGET_nios2 nios2-zephyr-elf)
+set(CROSS_COMPILE_TARGET_riscv32 riscv32-zephyr-elf)
+set(CROSS_COMPILE_TARGET_mips mipsel-zephyr-elf)
+set(CROSS_COMPILE_TARGET_xtensa xtensa-zephyr-elf)
+
 
 set(CROSS_COMPILE_TARGET     ${CROSS_COMPILE_TARGET_${ARCH}})
 set(SYSROOT_TARGET           ${CROSS_COMPILE_TARGET})

